sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -498,7 +498,9 @@ object EliminateDistinct extends Rule[LogicalPlan] {
   override def apply(plan: LogicalPlan): LogicalPlan = plan.transformWithPruning(
     _.containsPattern(AGGREGATE)) {
     case agg: Aggregate =>
-      agg.transformExpressionsWithPruning(_.containsPattern(AGGREGATE_EXPRESSION)) {
+      val isDistinctLiteral = isDistinctLiteralAggregate(agg)
+      var replaceDistinctAggregate = false
+      val newAgg = agg.transformExpressionsWithPruning(_.containsPattern(AGGREGATE_EXPRESSION)) {
```

Comment:
I don't think this is the right rule to put the rewrite, as this is an optimization rule. Users can skip it and Spark must still return the right result.

Reply:
you're right, but I think I'll close this PR anyways as it's now irrelevant